src/error.rs

```diff
@@ -29,3 +31,22 @@ impl From<jsonwebtoken::errors::Error> for JwksClientError {
         Self::Error(Arc::new(error.into()))
     }
 }
+
+impl JwksClientError {
+    pub fn is_jwt_expired(&self) -> bool {
```

Comment:
What about create a new error? Like `ExpiredJwt`

Reply:
Maybe will ends up a bit strange, you will have a `JsonWebToken` that contains all `jsonwebtoken::errors::Error` even the `ExpiredSignature` one, and another variant that will be returnet only in case of `ExpiredSignature`, the other option will be remap all the `ErrorKind` from jsonwebtoken 😓